### DISCUSSION TITLE
Is there a CI job that we can see all unit tests list in category?

### DISCUSSION THREAD

**Discussion by anatolianserdar** [openssl/openssl#26225] [Q&A] [answered]
I'm wondering if there is a Continuous Integration (CI) job available where we can view the complete list of unit tests. 

If anyone knows how to access this information or if there's an existing job to display all the unit tests, I would appreciate your guidance.

Thank you
Serdar

**Comment by nhorman** [comment]
depends on what exactly you mean by "all".  You can list all our unit tests locally by running:
```
make TESTS=list test
```

our CI jobs typically run all the unit tests we have under different configurations (different features enabled/different arches, etc).  for a list of that, you can take a look at the files in the .github/workflows directory

**Comment by anatolianserdar** [comment]
I was considering finding jobs or pipelines that list checked items, possibly under this link:
https://salsa.debian.org/debian/openssl/-/jobs.

More specifically, I am interested in the category related to RNG function tests.

**Comment by nhorman** [comment, marked as answer]
sure, you can all CI runs under the actions tab for a given project.  If you want them correlated by pull request, you can see them on that pull requests page.  for example here:
https://github.com/openssl/openssl/pull/26199

Each CI pipeline has a details link that shows the stages that that CI test runs.  for example:
https://github.com/openssl/openssl/actions/runs/12377920768/job/34548600997?pr=26199

Our CI runs are generally divided into the following stages:
*setup
*checkout
*configure
*config dump
*make (build)
*test
*post-run
*cleanup

The test stage typically runs our unit tests (identical to what you get when you run make test locally.  The test harness produces logs for each test run and its output.  for rng tests you can search for the term "rand" and see the unit tests and output that we run
